frontend/src/features/documents/documentUpload/DocumentUploadView.tsx

```diff
@@ -137,9 +145,20 @@ const DocumentUploadView: React.FunctionComponent<IDocumentUploadViewProps> = pr
                     <Input
                       field={value.metadata_type?.name || ''}
                       required={value.required === true}
+                      onChange={event => {
+                        if (value.metadata_type?.id) {
+                          props.documentMetadata[value.metadata_type?.id.toString()] =
```

Comment:
we should not modify the prop, instead we should use the formik field or a local variable to set it

Reply:
Changed to use formik field value